dunwoody_disaster/CharacterFactory.py

```diff
@@ -35,30 +41,33 @@ def __init__(self):
         self.defenses = {}
 
     def set_health(self, health: int):
-        self.curHealth = health
+        if health <= 0:
+            health = 0
+        else:
+            self.curHealth = health
```

Comment:
If we want to guard we could do something like 
```suggestion
        self.curHealth = min(self.maxHealth, max(0, health))
```
or we could panic
```suggestion
        if health < 0 or health > self.maxHealth:
            raise Exception('Invalid health argument')
```

Reply:
This goes for `set_magic` and `set_stamina` as well.